### PR TITLE
Use ret_num_trees for check integrity return

### PR DESCRIPTION
**Pull request by benjeffery**
Similarly to #1513

### CODE REVIEW

**Comment by jeromekelleher** [comment]
I'm not sure about this, because check_integrity only returns the number of trees **if we ask it to**. So, in most of these cases, the variable name will be actively misleading. I think we should use ``ret_num_trees`` in the cases where it is computing the number of trees, and cast to int in all the other cases, with a comment to the effect of "Casting to int is safe here because we do not compute the number of trees".

**Comment by jeromekelleher** [comment]
It was an unfortunate API choice, to be honest, in retrospect maybe we should have added another function or something.

**Comment by benjeffery** [comment]
> I'm not sure about this, because check_integrity only returns the number of trees **if we ask it to**. So, in most of these cases, the variable name will be actively misleading. I think we should use `ret_num_trees` in the cases where it is computing the number of trees, and cast to int in all the other cases, with a comment to the effect of "Casting to int is safe here because we do not compute the number of trees".

Ah yes, I had forgotten that the number of trees is not always returned.

**Comment by benjeffery** [comment]
@jeromekelleher now casting in all calls, except where TSK_CHECK_TREES is specified, where we use `ret_num_trees`

File: c/tests/test_minimal_cpp.cpp [hunk 1]
```diff
@@ -168,7 +168,9 @@ test_edge_sorting()
     assert(tables.edges.num_rows == (tsk_size_t) n - 1);
 
     /* Make sure the edges are unsorted */
-    ret = tsk_table_collection_check_integrity(&tables, TSK_CHECK_EDGE_ORDERING);
+    /* Not calling TSK_CHECK_TREES so casting is safe */
+    ret = (tsk_id_t) tsk_table_collection_check_integrity(
```
Comment (jeromekelleher) [inline c/tests/test_minimal_cpp.cpp:172]: We don't need to cast here, ret is declared as a tsk_id_t and the return value is already tsk_id_t. I think we should change the type of ``ret`` to int here (I probably changed this, before I realised it was a mistake to change ``ret`` values to tsk_id_t

**Review by jeromekelleher — COMMENTED** [review]
LGTM, but the C++ version is wrong

File: c/tests/test_minimal_cpp.cpp (hunk 1 above)
Comment (benjeffery) [reply c/tests/test_minimal_cpp.cpp]: Good catch, I've made `ret` an `int` here.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**